app/controllers/home_controller.rb

```diff
@@ -0,0 +1,20 @@
+class HomeController < ApplicationController
+  def index
+  end
+
+  def buzify
+    if (params[:number].present? && (params[:number].to_i.is_a? Integer))
```

Comment:
hey Fraz, while this code is performing correct output I don't think it belongs in the controller. Here is why :

- Business logic does not really belong in the controller. Controller is really meant to receive the request and delegate it to the proper resource to execute (resource can be a service/model etc). This keeps the MVC paradigm separate in the right way by encouraging loose coupling in your app.
- This business logic should really be inside some kind of service (a non active record class). The reason for moving this inside of service, is that you could then re-use this code somewhere else other than controller. Let's say you app were to implement some kind of messaging protocol, and you needed to determine the fizz buzzness there too. In that case you could re-use the fizz buzz service again without duplicating code.
- Another reason to extract this into service could be that you could test the service on its own, without involving the controller. In the controller test you should be really be testing response codes and body. All underlying code should be stubbed, therefore avoiding duplicate testing in both layers.

Having said this can you extract this into a service and write test for it. And call the service object from the controller. Also in case of non-integer input, you can raise an error. Then in the controller you can respond with 422 if error is raised.